src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/ObjectReaders.BoundAttributeParameterReader.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Serialization;
+
+internal static partial class ObjectReaders
+{
+    private record struct BoundAttributeParameterReader(BoundAttributeParameterDescriptorBuilder Builder)
+    {
+        public static readonly PropertyMap<BoundAttributeParameterReader> PropertyMap = new(
+            new(nameof(BoundAttributeParameterDescriptor.Kind), ReadKind),
+            new(nameof(BoundAttributeParameterDescriptor.Name), ReadName),
+            new(nameof(BoundAttributeParameterDescriptor.TypeName), ReadTypeName),
+            new(nameof(BoundAttributeParameterDescriptor.IsEnum), ReadIsEnum),
+            new(nameof(BoundAttributeParameterDescriptor.Documentation), ReadDocumentation),
+            new(nameof(BoundAttributeParameterDescriptor.Metadata), ReadMetadata),
+            new(nameof(BoundAttributeParameterDescriptor.Diagnostics), ReadDiagnostics));
+
+        private static void ReadKind(JsonDataReader reader, ref BoundAttributeParameterReader arg)
+        {
+            // In old serialized files, Kind might appear, though it isn't meaningful.
```

Comment:
If kind isn't there, do we do the right thing or are we throwing away data?

Reply:
We're throwing away data, but we need to handle it. It's unfortunate but it'll be going away with newer versions of the serialization format.